ansible_base/rbac/models.py

```diff
@@ -44,7 +44,38 @@ def __str__(self):
         return f"<{self.__class__.__name__}: {self.codename}>"
 
 
+class ManagedRoleFromSetting:
+    def __init__(self, role_name):
+        super().__init__()
+        self.role_name = role_name
+
+    def __get__(self, obj, objtype=None):
+        if obj is None:
+            return self
+        if self.role_name not in obj._cache:
+            obj._cache[self.role_name] = RoleDefinition.objects.filter(name=self.role_name).first()
+        return obj._cache[self.role_name]
```

Comment:
If someone deleted the role "Organization Admin" and this caches None what would happen in the code above? Would we get a NoneType has no attribute give_permission? Should we have a dummy object incase the objects.filter returns None that has the methods like give_permission/remove_permission but they are no-ops?

Reply:
I am thinking about role based authenticator_maps here. If we have a map that uses a role that gets deleted (either through a migration or because it was a custom role and it was no longer needed) it could still be in the authenticator_map. So we would either need to:

1. Cascade the delete to authenticator_maps (which feels wrong)
2. Prevent the deletion if the role is tied to an authenticator_map (which might be a pain for upgrades)
3. Modify claims.py to check to see if `RoleDefinition.objects.managed.org_member` returns None so we don't try and call give_permission/remove_permission. 

But this can be tackled later.